### ISSUE TITLE
print() signature changed in 0.14.0

### ISSUE THREAD

**Issue by ericsnowcurrently** [davidhalter/jedi#1358]
Here's what I ran (using Python 3.6):
```python
import jedi
script = jedi.Script('print(a, b, c, sep='')', 1, 9, 'script.py')
sig = script.call_signatures()[0]
for param in sig.params:
    print(param)
```

Using 0.14.0, I get the following output:
```
<Definition full_name=None, description='param *values: object'>
<Definition full_name=None, description='param sep: Text = ...'>
<Definition full_name=None, description='param end: Text = ...'>
<Definition full_name=None, description='param file: Optional[_Writer] = ...'>
<Definition full_name=None, description='param flush: bool = ...'>
```
Using 0.13.3, I get the following output:
```
<Definition param value>
<Definition param ...>
<Definition param sep>
<Definition param end>
<Definition param file>
<Definition param flush>
```

So the first param changed from "value" to "values" and the second param ("...") in 0.13.3 went away.  Also, it doesn't seem to match `help(print)` any longer: `print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)`.

**Comment by ericsnowcurrently** [comment]
Part of the problem here is that `sig.index` is confusing (in either version).  In the case of print I would expect all positional args to result in `sig.index` of 0 and a "sep=" arg to result in 1.  In the code above, that would be columns 6-18 and column 19, respectively.  Currently, if you move the column to 11 then `sig.index` changes to 2, which doesn't make a lot of sense.

**Comment by davidhalter** [comment]
100% agree that sig.index is still very buggy. I'll try to keep this open for that.

Other than that I would say that the new signature is actually more precise. It doesn't match `help(print)`, because that's not where we get the signature from. We get the signature from https://github.com/python/typeshed/blob/master/stdlib/2and3/builtins.pyi#L1335. I would not consider this to be a bug at all (appart from index).

**Comment by ericsnowcurrently** [comment]
Thanks for clarifying about typeshed, @davidhalter.  I've opened #1363 to cover the issue with the param index.

As to the precision of the signature, I get what you're saying.  However, there isn't anything about params currently to indicate that param 0 (in the case of print) is `*args`.  The name is "values".  This would be resolved with #1361, so I think we're good here.  Thanks!
